src/main/java/org/folio/linked/data/integration/kafka/sender/search/WorkCreateMessageSender.java

```diff
@@ -1,74 +1,75 @@
 import org.folio.linked.data.integration.kafka.sender.CreateMessageSender;
 import org.folio.linked.data.mapper.kafka.search.KafkaSearchMessageMapper;
 import org.folio.linked.data.model.entity.Resource;
 import org.folio.linked.data.model.entity.event.ResourceIndexedEvent;
-import org.folio.linked.data.repo.ResourceRepository;
-import org.folio.marc4ld.util.ResourceKind;
 import org.folio.search.domain.dto.LinkedDataWork;
 import org.folio.search.domain.dto.ResourceIndexEvent;
 import org.folio.spring.tools.kafka.FolioMessageProducer;
 import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Lazy;
 import org.springframework.context.annotation.Profile;
 import org.springframework.stereotype.Service;
 
 @Log4j2
 @Service
 @Profile(FOLIO_PROFILE)
-@RequiredArgsConstructor
 public class WorkCreateMessageSender implements CreateMessageSender {
 
   @Qualifier("bibliographicMessageProducer")
   private final FolioMessageProducer<ResourceIndexEvent> bibliographicMessageProducer;
   private final KafkaSearchMessageMapper<LinkedDataWork> searchBibliographicMessageMapper;
-  private final ApplicationEventPublisher eventPublisher;
-  private final ResourceRepository resourceRepository;
+  private final ResourceModificationEventListener eventListener;
+  private final WorkUpdateMessageSender workUpdateMessageSender;
+
+  public WorkCreateMessageSender(FolioMessageProducer<ResourceIndexEvent> bibliographicMessageProducer,
+                                 KafkaSearchMessageMapper<LinkedDataWork> searchBibliographicMessageMapper,
+                                 @Lazy ResourceModificationEventListener eventListener,
+                                 WorkUpdateMessageSender workUpdateMessageSender) {
+    this.bibliographicMessageProducer = bibliographicMessageProducer;
+    this.searchBibliographicMessageMapper = searchBibliographicMessageMapper;
+    this.eventListener = eventListener;
+    this.workUpdateMessageSender = workUpdateMessageSender;
+  }
 
   @Override
   public Collection<Resource> apply(Resource resource) {
-    return Stream.of(resource)
-      .filter(this::test)
-      .toList();
+    if (resource.isOfType(WORK)) {
+      return singletonList(resource);
+    }
+    if (resource.isOfType(INSTANCE)) {
+      triggerParentWorkUpdate(resource);
+    }
+    return emptyList();
   }
 
-  //TODO refactoring to extract resources
-  private boolean test(Resource resource) {
-    return ResourceKind.BIBLIOGRAPHIC
-      .stream()
-      .anyMatch(resource::isOfType);
+  private void triggerParentWorkUpdate(Resource instance) {
+    extractWork(instance)
+      .ifPresentOrElse(work -> {
+        log.info("Instance [id {}] creation triggered parent Work [id {}] index update",
+          instance.getId(), work.getId());
+        workUpdateMessageSender.produce(work);
```

Comment:
minor
for me it looks like "work event update,  not a directly call
why you don't use event here?
like eventListener::afterUpdate

Reply:
Earlier I thought like that too.
Then, I realized that it's wrong design. Let me explain:
1) We have an event occurred in our system: Resource is created.
2) Event subscribers handle it accordingly to their needs.
3) One of subscribers needs to do additional action - update another resource's search index. It shouldn't be a new event! It's logical subpart of the initial event. It shouldn't be passed to all subscribers again - all of them are  already finished their jobs regarding initial event.

Reply:
discussed, agreed - ok as is